probe-rs/src/probe/cmsisdap/tools.rs

```diff
@@ -62,8 +62,8 @@ fn get_cmsisdap_info(device: &Device<rusb::Context>) -> Option<DebugProbeInfo> {
         .read_serial_number_string(language, &d_desc, timeout)
         .ok();
 
-    // Most CMSIS-DAP probes say "CMSIS-DAP" in their product string.
-    let cmsis_dap_product = prod_str.contains("CMSIS-DAP");
+    // Most CMSIS-DAP probes say "CMSIS-DAP" or "CMSIS_DAP" in their product string.
+    let cmsis_dap_product = prod_str.contains("CMSIS-DAP") || prod_str.contains("CMSIS_DAP");
```

Comment:
Apologize for drive-by comment but, perhaps worth moving the comparsion into a free function ? Would clean up duplication here and below
```
fn is_cmsis_dap(id :& str ) -> bool {
    id.contains("CMSIS-DAP") || id.contains("CMSIS_DAP")
}
```

Reply:
Hey, thanks for bringing that up! That seems like a good idea, it would make it easier to add more spellings of `cmsis dap` in the future. Would also be useful we end up moving to a regex in the future. I am gonna go ahead and implement that.